u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/u2f/U2fAttestation.java

```diff
@@ -47,14 +50,15 @@
    *      OCTET STRING (1 elem)
    *        BIT STRING (4 bits) 1101
    *
-   * @param cert the certificate to parse for extension
+   * @param certChain the certificate to parse for extension
    * @return the supported transports as a List of HardwareFeatures or null if no extension
    * was found
    * @throws CertificateParsingException
    */
-  public static U2fAttestation Parse(X509Certificate cert) throws CertificateParsingException {
+  public static U2fAttestation Parse(X509Certificate[] certChain)
+      throws CertificateParsingException {
     ASN1OctetString extValue =
-        X509ExtensionParsingUtil.extractExtensionValue(cert, TRANSPORT_EXTENSION_OID);
+        X509ExtensionParsingUtil.extractExtensionValue(certChain[0], TRANSPORT_EXTENSION_OID);
```

Comment:
yet you only really parse the first (implicitly, the leaf) in the chain. Why not have the caller just pass in the leaf for you, and leave this method as it is?